### ISSUE TITLE
Write a "check owner/group/permissions" script

### ISSUE THREAD

**Issue by bmcutler** [Submitty/Submitty#991]
Might do more...   check for existence of required files, specific configuration parameters, etc.

Can be used as a consistency/sanity check on the production server, especially helpful if we patch or upgrade mid-semester.

Can be used as part of automated system testing, to make sure we've populated the system with 'fake' data correctly.

**Comment by MasterOdin** [comment]
Could we get a print out of permissions of directories on the live server as the base for what "good" permissions are.

Should get:
all files + folders in /usr/local/submitty excluding GIT_* directories
all files + folders in /var/local/submitty except we only need one semester + one course and then only one homework and submission for it

**Comment by bmcutler** [comment]
Rather than learning from the live server, I think we should write down the 'rules', and test against the live server to debug the rules.
The create_course.sh script already contains some of the documentation of the design/rules.

**Comment by MasterOdin** [comment]
Fair enough.

I do think that for testing, we should be using the Path interface available to us from Python 3.4 onward.

So to get the a folder/file's permissions:
```
root@vagrant-ubuntu-trusty-64:/var/local/submitty/courses/s17/csci1000# python3
Python 3.4.3 (default, Nov 17 2016, 01:08:31)
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> p = Path('/usr/local/submitty/bin')
>>> p.owner()
'root'
>>> p.group()
'course_builders'
>>> for child in p.iterdir():
...     print(child.name + ":" + child.owner() + " " + child.group() + " " + oct(child.stat().st_mode))
...
regrade.sh:root course_builders 0o100550
grade_students.py:root root 0o100500
grade_students.sh:root hwcron 0o100550
grading_done.sh:root course_builders 0o100550
system_call_check.out:root course_builders 0o100550
fake_submit_button_press.sh:root root 0o100500
make_assignments_txt_file.py:hwcron course_builders 0o100550
build_config_upload.py:root hwcron 0o100550
build_homework_function.sh:hwcron course_builders 0o100550
get_version_details.py:root course_builders 0o100550
create_course.sh:root root 0o100500
setcsvfields:root root 0o100500
insert_database_version_data.py:hwcron hwcron 0o100500
build_course.sh:root root 0o100500
read_iclicker_ids.py:root course_builders 0o100550
grade_students__results_history.py:root hwcron 0o100550
untrusted_execute:root hwcron 0o104550
```

**Comment by bmcutler** [comment]
started work in check_owner_etc branch:
https://github.com/Submitty/Submitty/tree/check_owner_etc
